### DISCUSSION TITLE
withInitialState() method

### DISCUSSION THREAD

**Discussion by Grafikart** [statelyai/xstate#3449] [Ideas]
For testing purpose it would be great to be able to set the initial state with a simple method. We already have `withConfig()` and `withContext()` but there is no way to change the starting state easily.

The current solution I have

```ts
export function makeGame (state: MachineStates, context: Partial<MachineContext> = {}): InterpreterFrom<typeof MyMachine> {
  const machine = interpret(
    MyMachine.withContext({
      ...MyMachineModel.initialContext,
      ...context
    })
  ).start()
  machine.state.value = state // This is not ideal
  return machine
}
```

Another solution would be to pass a state to the start method, but there is no easy way to create a state starting at a specific point :(

**Comment by ChrisShank** [comment]
You can pass an initial state the `start` i believe. https://xstate.js.org/docs/guides/states.html#persisting-state

**Reply by Grafikart** [reply]
This is the second solution I mentioned in my issue, but creating a `State`  is a bit complicated from scratch since the object has a lot of properties
